hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -408,6 +411,7 @@ private void doConnectToCluster() {
 
         // try the current cluster
         if (doConnectToCandidateCluster(currentContext, false)) {
+            connectionProcessListener.clusterConnectionSucceeded(currentContext.getClusterName());
```

Comment:
When failover is enabled, we can also connect to the cluster in the
```java
        if (clusterDiscoveryService.tryNextCluster(this::destroyCurrentClusterConnectionAndTryNextCluster)) {
            return;
        }
```

below. Should we also handle that?

Reply:
I intentionally omitted that for now, since MC doesn't support failover configs.